packages/katello/katello/hostname-change.rb

```diff
@@ -345,6 +345,53 @@ def run
       STDOUT.puts "updating hostname in foreman installer scenarios"
       self.run_cmd("sed -i -e 's/#{@old_hostname}/#{@new_hostname}/g' #{scenarios_path}/*.yaml")
 
+      if @scenario_answers['foreman_proxy']['dns']
+        STDOUT.puts 'gathering DNS data...'
+        local_ip = '127.0.0.1'
+        zone = @scenario_answers['foreman_proxy']['dns_zone']
+        old_fqdn = @old_hostname
+        soa_admin_domain = "root.#{zone}"
+        zone_file = "/var/named/dynamic/db.#{zone}"
+        zone_file_data = File.read(zone_file).split("\n")
+        a_record = zone_file_data.find { |line| line.include?(' A ') }
+        existing_ip = a_record.split(' A ').last if a_record
```

Comment:
will `existing_ip` not exist if there is no `a_record`? wouldn't it cause errors when it is referenced below if so?